src/Pipeline.php

```diff
@@ -29,13 +41,18 @@ public function pipe(callable $stage): PipelineInterface
         return $pipeline;
     }
 
-    public function process($payload)
+    public function process($payload, ...$params)
     {
+        if ($this->processor instanceof ParametersInterface) {
+            $this->processor->setParameters(...$params);
+            return $this->processor->process($payload, ...$this->stages);
+        }
+
         return $this->processor->process($payload, ...$this->stages);
```

Comment:
It might make more sense for this to be:

```php
return $this->processor->withParameters($params)->process(...$this->stages);
```

Doing so will change the signatures significantly though, and cause a major version bump.

Let me think about this for a couple of days.

Reply:
Yes, your variant is more elegant, but needs to bump the major version.
> Let me think about this for a couple of days.

ok.